apps/src/aichat/views/UserChatMessageEditor.tsx

```diff
@@ -16,14 +19,39 @@ const UserChatMessageEditor: React.FunctionComponent = () => {
     (state: {aichat: AichatState}) => state.aichat.isWaitingForChatResponse
   );
 
-  // TODO: If systemPrompt is undefined, handle this error case.
+  const userId: number = useSelector(
+    (state: {currentUser: CurrentUserState}) => state.currentUser.userId
+  );
+
+  const {currentLevelId, scriptId} = useSelector(
+    (state: {progress: ProgressState}) => state.progress
+  );
+
+  const channelId = getStandaloneProjectId();
+
   const dispatch = useAppDispatch();
+
   const handleSubmit = useCallback(() => {
     if (!isWaitingForChatResponse) {
-      dispatch(submitChatMessage(userMessage));
+      const chatContents: ChatContext = {
+        userMessage,
```

Comment:
Yeah, so technically we shouldn't be accessing redux state outside of React/redux code. We do have a lot of instances of that in older code, but in newer code we're generally trying to avoid it to enforce the separation between react/redux and non-react code.